### ISSUE TITLE
Decoding some grayscale AVIF to PNG fail

### ISSUE THREAD

**Issue by q3cpma** [AOMediaCodec/libavif#1385]
Here's the input file for reproduction: [test_gray_manga](https://github.com/AOMediaCodec/libavif/assets/40627987/fbc8273e-f3c8-45fc-9eb4-d376178cf9d6)

```
$ avifenc --version
Version: 0.10.1 (dav1d [dec]:1.1.0, aom [enc/dec]:v3.6.0)
$ magick identify test_gray_manga.png
test_gray_manga.png PNG 1809x2558 1809x2558+0+0 8-bit Grayscale Gray 256c 2.82477MiB 0.000u 0:00.000
$ avifenc -l -y 400 test_gray_manga.png test_gray_manga.avif
Successfully loaded: test_gray_manga.png
AVIF to be written: (Lossless)
 * Resolution     : 1809x2558
 * Bit Depth      : 8
 * Format         : YUV400
 * Alpha          : Not premultiplied
 * Range          : Full
 * Color Primaries: 2
 * Transfer Char. : 2
 * Matrix Coeffs. : 6
 * ICC Profile    : Present (544 bytes)
 * XMP Metadata   : Absent (0 bytes)
 * EXIF Metadata  : Absent (0 bytes)
 * Transformations: None
 * Progressive    : Unavailable
Encoding with AV1 codec 'aom' speed [6], color QP [0 (Lossless) <-> 0 (Lossless)], alpha QP [0 (Lossless) <-> 0 (Lossless)], tileRowsLog2 [0], tileColsLog2 [0], 1 worker thread(s), please wait...
Encoded successfully.
 * Color AV1 total size: 3376922 bytes
 * Alpha AV1 total size: 0 bytes
Wrote AVIF: test_gray_manga.avif
$ avifdec test_gray_manga.avif test_gray_manga.avif.png
Decoding with AV1 codec 'dav1d' (1 worker thread), please wait...
Image decoded: test_gray_manga.avif
Image details:
 * Resolution     : 1809x2558
 * Bit Depth      : 8
 * Format         : YUV400
 * Alpha          : Absent
 * Range          : Full
 * Color Primaries: 2
 * Transfer Char. : 2
 * Matrix Coeffs. : 6
 * ICC Profile    : Present (544 bytes)
 * XMP Metadata   : Absent (0 bytes)
 * EXIF Metadata  : Absent (0 bytes)
 * Transformations: None
 * Progressive    : Unavailable
libpng error: profile 'libavif': 'GRAY': Gray color space not permitted on RGB PNG
Error writing PNG: test_gray_manga.avif.png
```
Using Y4M output works fine.

Maybe related, but on another grayscale image (which decodes to PNG properly, though not a grayscale PNG), avifenc chooses different parameters (color primaries, transfer char. and matrix coeffs):
```
$ magick convert -colorspace gray rose: rose.png
$ magick identify rose.png
rose.png PNG 70x46 70x46+0+0 8-bit Grayscale Gray 256c 2445B 0.000u 0:00.000
$ avifenc -l -y 400 rose.png rose.avif
Successfully loaded: rose.png
AVIF to be written: (Lossless)
 * Resolution     : 70x46
 * Bit Depth      : 8
 * Format         : YUV400
 * Alpha          : Not premultiplied
 * Range          : Full
 * Color Primaries: 1
 * Transfer Char. : 13
 * Matrix Coeffs. : 6
 * ICC Profile    : Absent (0 bytes)
 * XMP Metadata   : Absent (0 bytes)
 * EXIF Metadata  : Absent (0 bytes)
 * Transformations: None
 * Progressive    : Unavailable
Encoding with AV1 codec 'aom' speed [6], color QP [0 (Lossless) <-> 0 (Lossless)], alpha QP [0 (Lossless) <-> 0 (Lossless)], tileRowsLog2 [0], tileColsLog2 [0], 1 worker thread(s), please wait...
Encoded successfully.
 * Color AV1 total size: 2446 bytes
 * Alpha AV1 total size: 0 bytes
Wrote AVIF: rose.avif
$ avifdec rose.avif rose.avif.png
Decoding with AV1 codec 'dav1d' (1 worker thread), please wait...
Image decoded: rose.avif
Image details:
 * Resolution     : 70x46
 * Bit Depth      : 8
 * Format         : YUV400
 * Alpha          : Absent
 * Range          : Full
 * Color Primaries: 1
 * Transfer Char. : 13
 * Matrix Coeffs. : 6
 * ICC Profile    : Absent (0 bytes)
 * XMP Metadata   : Absent (0 bytes)
 * EXIF Metadata  : Absent (0 bytes)
 * Transformations: None
 * Progressive    : Unavailable
Wrote PNG: rose.avif.png
$ magick identify rose.avif.png
rose.avif.png PNG 70x46 70x46+0+0 8-bit sRGB 5112B 0.000u 0:00.000
```

By the way, couldn't the -y 400 parameter be set automatically for grayscale input?

**Comment by q3cpma** [comment]
On Fri May 12, 2023 at 5:53 PM CEST, Vincent Rabaud wrote:
> HEAD does not allow 400 with lossless
Strange, as some pictures do seem to work.

> while 400 works and is set automatically for lossy.
> I am working on a patch to fix that and have 400 automatically chosen for lossless.
Thanks, that'll be welcome!

**Comment by tongyuantongyu** [comment]
Seems like a duplicate of #1123 and is already fixed in #1121. Please upgrade your `libavif` to `v0.11.0` or higher.
